gtda/diagrams/features.py

```diff
@@ -391,3 +392,190 @@ def transform(self, X, y=None):
+                 for dim in self.homology_dimensions_]
+        elif type(self.n_coefficients) == list:
+            if len(self.n_coefficients) != len(self.homology_dimensions_):
+                raise ValueError(f'n_coefficients has been passed as a list '
+                                 f'of length {len(self.n_coefficients)} while '
+                                 f'diagrams in `X` have '
+                                 f'{len(self.homology_dimensions_)} homology '
+                                 f'dimensions.')
+            self.n_coefficients_ = self.n_coefficients
+        else:
+            self.n_coefficients_ = \
+                [self.n_coefficients] * len(self.homology_dimensions_)
+
+        self._polynomial_function = \
+            _implemented_polynomial_recipes[self.polynomial_type]
+
+        return self
+
+    def _complex_polynomial(self, X, n_coefficients):
+        Xt = np.zeros(2 * n_coefficients,)
+        X = X[X[:, 0] != X[:, 1]]
+
+        roots = self._polynomial_function(X)
+        coefficients = np.poly(roots)
+
+        coefficients = np.array(coefficients[1:])
+        dimension = min(n_coefficients, coefficients.shape[0])
+        Xt[:dimension] = coefficients[:dimension].real
+        Xt[n_coefficients:n_coefficients + dimension] = \
+            coefficients[:dimension].imag
+
+        return Xt
+
+    def transform(self, X, y=None):
+        """Computes the real and imaginary parts of the complex vector of
+        coefficients for each diagram individually and concatenate the results.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features, 3)
+            Input data. Array of persistence diagrams, each a collection of
+            triples [b, d, q] representing persistent topological features
+            through their birth (b), death (d) and homology dimension (q).
+            It is important that, for each possible homology dimension, the
+            number of triples for which q equals that homology dimension is
+            constants across the entries of `X`.
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        Xt : ndarray of shape (n_samples, n_homology_dimensions * 2 \
+            * n_coefficients)
+            Polynomial coefficients: Real and imaginary parts of the complex
+            polynomial for each homology dimension of each diagrams in `X`.
+
+        """
+        check_is_fitted(self)
+        Xt = check_diagrams(X, copy=True)
+
+        Xt = Parallel(n_jobs=self.n_jobs)(
+            delayed(self._complex_polynomial)(
+                _subdiagrams(Xt[s], [dim], remove_dim=True),
+                self.n_coefficients_[d])
+            for s in range(len(X))
+            for d, dim in enumerate(self.homology_dimensions_))
+        Xt = np.concatenate(Xt).reshape(len(X), -1)
```

Comment:
This looks quite different from the more involved reshapings we do in the rest of this module. @gtauzin just checking whether this is deliberate and has been tested (I'll double check asap).

Reply:
Indeed, it's a bit different, also because `subdiagrams_` accepts collections of diagrams, but `complex_polynomial` takes a single diagram.